ratpack-exec/src/main/java/ratpack/exec/internal/DefaultExecution.java

```diff
@@ -631,12 +651,12 @@ void error(Throwable throwable) {
     }
   }
 
-  private static final class Ref implements ExecutionRef {
+  static final class Ref implements ExecutionRef {
```

Comment:
This is all a bit to heavy handed just so we can get the `nextChild()` method exposed to the correct place. 
I think it makes more sense to elevate `nextChild()` onto the `ExecutionRef` interface. I'd also probably call it something else (`getAndIncrementChildCount()` is what it is but that's verbose).

Reply:
Removed in my PR against @tomdcc's fork

Reply:
I did leave this as `nextChild()` for the moment, but would certainly be open to whether you just want me to refactor it to `getAndIncrementChildCount()`